### ISSUE TITLE
k0smotron operator falls into CrashLoopBackOff

### ISSUE THREAD

**Issue by aerfio** [k0sproject/k0smotron#659]
Reproduction steps:
1. Create an empty kind cluster (or use any other tool):
	```
	kind create cluster --wait=5m --name k0smotron
	```
2. Install k0smotron following official instruction: https://docs.k0smotron.io/v1.0.2/install/
	```
	kubectl apply --server-side=true -f https://docs.k0smotron.io/v1.0.2/install.yaml
	```
3. Wait several minutes and see k0smotron crashing:
	```
	➜ kubectl get pod -n k0smotron
	NAME                                            READY   STATUS    RESTARTS      AGE
	k0smotron-controller-manager-64f9c7dc58-dcpc9   2/2     Running   3 (56s ago)   7m54s
	```

The pod logs: 
```log
2024-07-29T13:49:43Z    ERROR   controller-runtime.source.EventHandler  failed to get informer from cache       {"error": "failed to get API group resources: unable to retrieve the complete list of server APIs: cluster.x-k8s.io/v1beta1: the server could not find the requested resource"}
sigs.k8s.io/controller-runtime/pkg/internal/source.(*Kind).Start.func1.1
        /go/pkg/mod/sigs.k8s.io/controller-runtime@v0.16.5/pkg/internal/source/kind.go:68
k8s.io/apimachinery/pkg/util/wait.loopConditionUntilContext.func2
        /go/pkg/mod/k8s.io/apimachinery@v0.28.4/pkg/util/wait/loop.go:73
k8s.io/apimachinery/pkg/util/wait.loopConditionUntilContext
        /go/pkg/mod/k8s.io/apimachinery@v0.28.4/pkg/util/wait/loop.go:74
k8s.io/apimachinery/pkg/util/wait.PollUntilContextCancel
        /go/pkg/mod/k8s.io/apimachinery@v0.28.4/pkg/util/wait/poll.go:33
sigs.k8s.io/controller-runtime/pkg/internal/source.(*Kind).Start.func1
        /go/pkg/mod/sigs.k8s.io/controller-runtime@v0.16.5/pkg/internal/source/kind.go:56
2024-07-29T13:49:53Z    ERROR   Could not wait for Cache to sync        {"controller": "machine", "controllerGroup": "cluster.x-k8s.io", "controllerKind": "Machine", "error": "failed to wait for machine caches to sync: timed out waiting for cache to be synced for Kind *v1beta1.Machine"}
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.1
        /go/pkg/mod/sigs.k8s.io/controller-runtime@v0.16.5/pkg/internal/controller/controller.go:203
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2
        /go/pkg/mod/sigs.k8s.io/controller-runtime@v0.16.5/pkg/internal/controller/controller.go:208
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start
        /go/pkg/mod/sigs.k8s.io/controller-runtime@v0.16.5/pkg/internal/controller/controller.go:234
sigs.k8s.io/controller-runtime/pkg/manager.(*runnableGroup).reconcile.func1
        /go/pkg/mod/sigs.k8s.io/controller-runtime@v0.16.5/pkg/manager/runnable_group.go:223
2024-07-29T13:49:53Z    INFO    Stopping and waiting for non leader election runnables
2024-07-29T13:49:53Z    INFO    Stopping and waiting for leader election runnables
2024-07-29T13:49:53Z    INFO    Shutdown signal received, waiting for all workers to finish     {"controller": "k0smotroncontrolplane", "controllerGroup": "controlplane.cluster.x-k8s.io", "controllerKind": "K0smotronControlPlane"}
2024-07-29T13:49:53Z    INFO    Shutdown signal received, waiting for all workers to finish     {"controller": "remotecluster", "controllerGroup": "infrastructure.cluster.x-k8s.io", "controllerKind": "RemoteCluster"}
2024-07-29T13:49:53Z    INFO    Shutdown signal received, waiting for all workers to finish     {"controller": "remotemachine", "controllerGroup": "infrastructure.cluster.x-k8s.io", "controllerKind": "RemoteMachine"}
2024-07-29T13:49:53Z    INFO    Shutdown signal received, waiting for all workers to finish     {"controller": "k0scontrolplane", "controllerGroup": "controlplane.cluster.x-k8s.io", "controllerKind": "K0sControlPlane"}
2024-07-29T13:49:53Z    INFO    Shutdown signal received, waiting for all workers to finish     {"controller": "jointokenrequest", "controllerGroup": "k0smotron.io", "controllerKind": "JoinTokenRequest"}
2024-07-29T13:49:53Z    INFO    Shutdown signal received, waiting for all workers to finish     {"controller": "k0scontrollerconfig", "controllerGroup": "bootstrap.cluster.x-k8s.io", "controllerKind": "K0sControllerConfig"}
2024-07-29T13:49:53Z    INFO    Shutdown signal received, waiting for all workers to finish     {"controller": "k0sworkerconfig", "controllerGroup": "bootstrap.cluster.x-k8s.io", "controllerKind": "K0sWorkerConfig"}
2024-07-29T13:49:53Z    INFO    Shutdown signal received, waiting for all workers to finish     {"controller": "cluster", "controllerGroup": "k0smotron.io", "controllerKind": "Cluster"}
2024-07-29T13:49:53Z    INFO    All workers finished    {"controller": "remotemachine", "controllerGroup": "infrastructure.cluster.x-k8s.io", "controllerKind": "RemoteMachine"}
2024-07-29T13:49:53Z    INFO    All workers finished    {"controller": "k0smotroncontrolplane", "controllerGroup": "controlplane.cluster.x-k8s.io", "controllerKind": "K0smotronControlPlane"}
2024-07-29T13:49:53Z    INFO    All workers finished    {"controller": "remotecluster", "controllerGroup": "infrastructure.cluster.x-k8s.io", "controllerKind": "RemoteCluster"}
2024-07-29T13:49:53Z    INFO    All workers finished    {"controller": "k0scontrolplane", "controllerGroup": "controlplane.cluster.x-k8s.io", "controllerKind": "K0sControlPlane"}
2024-07-29T13:49:53Z    INFO    All workers finished    {"controller": "k0scontrollerconfig", "controllerGroup": "bootstrap.cluster.x-k8s.io", "controllerKind": "K0sControllerConfig"}
2024-07-29T13:49:53Z    INFO    All workers finished    {"controller": "jointokenrequest", "controllerGroup": "k0smotron.io", "controllerKind": "JoinTokenRequest"}
2024-07-29T13:49:53Z    INFO    All workers finished    {"controller": "k0sworkerconfig", "controllerGroup": "bootstrap.cluster.x-k8s.io", "controllerKind": "K0sWorkerConfig"}
2024-07-29T13:49:53Z    INFO    All workers finished    {"controller": "cluster", "controllerGroup": "k0smotron.io", "controllerKind": "Cluster"}
2024-07-29T13:49:53Z    INFO    Stopping and waiting for caches
2024-07-29T13:49:53Z    INFO    Stopping and waiting for webhooks
2024-07-29T13:49:53Z    INFO    Stopping and waiting for HTTP servers
2024-07-29T13:49:53Z    INFO    controller-runtime.metrics      Shutting down metrics server with timeout of 1 minute
2024-07-29T13:49:53Z    INFO    shutting down server    {"kind": "health probe", "addr": "[::]:8081"}
2024-07-29T13:49:53Z    INFO    Wait completed, proceeding to shutdown the manager
2024-07-29T13:49:53Z    ERROR   setup   problem running manager {"error": "failed to wait for machine caches to sync: timed out waiting for cache to be synced for Kind *v1beta1.Machine"}
main.main
        /workspace/cmd/main.go:247
runtime.main
        /usr/local/go/src/runtime/proc.go:267
```

**Comment by makhov** [comment]
Thanks for the report. We fixed the issue and published k0smotron v1.0.3 with the fix

**Comment by EKami** [comment]
I have exactly the same issue but with k0smotron 1.2.0:
```
❯ k logs k0smotron-controller-manager-864bc75cf7-zxxjc -n k0smotron
2024-12-19T10:05:43Z    INFO    CAPI crds are not installed yet, skipping initializing providerID controller
2024-12-19T10:05:43Z    INFO    setup   starting manager
2024-12-19T10:05:43Z    INFO    controller-runtime.metrics      Starting metrics server
2024-12-19T10:05:43Z    INFO    controller-runtime.metrics      Serving metrics server  {"bindAddress": "127.0.0.1:8080", "secure": false}
2024-12-19T10:05:43Z    INFO    starting server {"kind": "health probe", "addr": "[::]:8081"}
I1219 10:05:43.613962       1 leaderelection.go:250] attempting to acquire leader lease k0smotron/d41d8cd98f00b204e9800998ecf8427e.k0smotron.io...
I1219 10:05:59.139702       1 leaderelection.go:260] successfully acquired lease k0smotron/d41d8cd98f00b204e9800998ecf8427e.k0smotron.io
2024-12-19T10:05:59Z    DEBUG   events  k0smotron-controller-manager-864bc75cf7-zxxjc_b965ead6-c9d5-40c9-81f2-96296b117625 became leader        {"type": "Normal", "object": {"kind":"Lease","namespace":"k0smotron","name":"d41d8cd98f00b204e9800998ecf8427e.k0smotron.io","uid":"0979fca4-4770-4151-9335-6fe77fcce810","apiVersion":"coordination.k8s.io/v1","resourceVersion":"13139"}, "reason": "LeaderElection"}
2024-12-19T10:05:59Z    INFO    Starting EventSource    {"controller": "cluster", "controllerGroup": "k0smotron.io", "controllerKind": "Cluster", "source": "kind source: *v1beta1.Cluster"}
2024-12-19T10:05:59Z    INFO    Starting EventSource    {"controller": "cluster", "controllerGroup": "k0smotron.io", "controllerKind": "Cluster", "source": "kind source: *v1.StatefulSet"}
2024-12-19T10:05:59Z    INFO    Starting Controller     {"controller": "cluster", "controllerGroup": "k0smotron.io", "controllerKind": "Cluster"}
2024-12-19T10:05:59Z    INFO    Starting EventSource    {"controller": "remotemachine", "controllerGroup": "infrastructure.cluster.x-k8s.io", "controllerKind": "RemoteMachine", "source": "kind source: *v1beta1.RemoteMachine"}
2024-12-19T10:05:59Z    INFO    Starting Controller     {"controller": "remotemachine", "controllerGroup": "infrastructure.cluster.x-k8s.io", "controllerKind": "RemoteMachine"}
2024-12-19T10:05:59Z    INFO    Starting EventSource    {"controller": "k0sworkerconfig", "controllerGroup": "bootstrap.cluster.x-k8s.io", "controllerKind": "K0sWorkerConfig", "source": "kind source: *v1beta1.K0sWorkerConfig"}
2024-12-19T10:05:59Z    INFO    Starting Controller     {"controller": "k0sworkerconfig", "controllerGroup": "bootstrap.cluster.x-k8s.io", "controllerKind": "K0sWorkerConfig"}
2024-12-19T10:05:59Z    INFO    Starting EventSource    {"controller": "k0smotroncontrolplane", "controllerGroup": "controlplane.cluster.x-k8s.io", "controllerKind": "K0smotronControlPlane", "source": "kind source: *v1beta1.K0smotronControlPlane"}
2024-12-19T10:05:59Z    INFO    Starting EventSource    {"controller": "k0smotroncontrolplane", "controllerGroup": "controlplane.cluster.x-k8s.io", "controllerKind": "K0smotronControlPlane", "source": "kind source: *v1beta1.Cluster"}
2024-12-19T10:05:59Z    INFO    Starting Controller     {"controller": "k0smotroncontrolplane", "controllerGroup": "controlplane.cluster.x-k8s.io", "controllerKind": "K0smotronControlPlane"}
2024-12-19T10:05:59Z    INFO    Starting EventSource    {"controller": "k0scontrolplane", "controllerGroup": "controlplane.cluster.x-k8s.io", "controllerKind": "K0sControlPlane", "source": "kind source: *v1beta1.K0sControlPlane"}
2024-12-19T10:05:59Z    INFO    Starting EventSource    {"controller": "k0scontrolplane", "controllerGroup": "controlplane.cluster.x-k8s.io", "controllerKind": "K0sControlPlane", "source": "kind source: *v1beta1.Machine"}
2024-12-19T10:05:59Z    INFO    Starting Controller     {"controller": "k0scontrolplane", "controllerGroup": "controlplane.cluster.x-k8s.io", "controllerKind": "K0sControlPlane"}
2024-12-19T10:05:59Z    INFO    Starting EventSource    {"controller": "jointokenrequest", "controllerGroup": "k0smotron.io", "controllerKind": "JoinTokenRequest", "source": "kind source: *v1beta1.JoinTokenRequest"}
2024-12-19T10:05:59Z    INFO    Starting Controller     {"controller": "jointokenrequest", "controllerGroup": "k0smotron.io", "controllerKind": "JoinTokenRequest"}
2024-12-19T10:05:59Z    INFO    Starting EventSource    {"controller": "k0scontrollerconfig", "controllerGroup": "bootstrap.cluster.x-k8s.io", "controllerKind": "K0sControllerConfig", "source": "kind source: *v1beta1.K0sControllerConfig"}
2024-12-19T10:05:59Z    INFO    Starting Controller     {"controller": "k0scontrollerconfig", "controllerGroup": "bootstrap.cluster.x-k8s.io", "controllerKind": "K0sControllerConfig"}
2024-12-19T10:05:59Z    INFO    Starting EventSource    {"controller": "remotecluster", "controllerGroup": "infrastructure.cluster.x-k8s.io", "controllerKind": "RemoteCluster", "source": "kind source: *v1beta1.RemoteCluster"}
2024-12-19T10:05:59Z    INFO    Starting Controller     {"controller": "remotecluster", "controllerGroup": "infrastructure.cluster.x-k8s.io", "controllerKind": "RemoteCluster"}

sigs.k8s.io/controller-runtime/pkg/internal/source.(*Kind).Start.func1
        /go/pkg/mod/sigs.k8s.io/controller-runtime@v0.16.5/pkg/internal/source/kind.go:56
2024-12-19T10:05:59Z    ERROR   controller-runtime.source.EventHandler  failed to get informer from cache       {"error": "failed to get API group resources: unable to retrieve the complete list of server APIs: cluster.x-k8s.io/v1beta1: the server could not find the requested resource"}
sigs.k8s.io/controller-runtime/pkg/internal/source.(*Kind).Start.func1.1
        /go/pkg/mod/sigs.k8s.io/controller-runtime@v0.16.5/pkg/internal/source/kind.go:68
k8s.io/apimachinery/pkg/util/wait.loopConditionUntilContext.func2
        /go/pkg/mod/k8s.io/apimachinery@v0.28.4/pkg/util/wait/loop.go:73
k8s.io/apimachinery/pkg/util/wait.loopConditionUntilContext
        /go/pkg/mod/k8s.io/apimachinery@v0.28.4/pkg/util/wait/loop.go:74
k8s.io/apimachinery/pkg/util/wait.PollUntilContextCancel
        /go/pkg/mod/k8s.io/apimachinery@v0.28.4/pkg/util/wait/poll.go:33
sigs.k8s.io/controller-runtime/pkg/internal/source.(*Kind).Start.func1
        /go/pkg/mod/sigs.k8s.io/controller-runtime@v0.16.5/pkg/internal/source/kind.go:56
2024-12-19T10:05:59Z    INFO    Starting workers        {"controller": "jointokenrequest", "controllerGroup": "k0smotron.io", "controllerKind": "JoinTokenRequest", "worker count": 1}
2024-12-19T10:05:59Z    INFO    Starting workers        {"controller": "k0scontrollerconfig", "controllerGroup": "bootstrap.cluster.x-k8s.io", "controllerKind": "K0sControllerConfig", "worker count": 1}
2024-12-19T10:05:59Z    INFO    Starting workers        {"controller": "k0smotroncontrolplane", "controllerGroup": "controlplane.cluster.x-k8s.io", "controllerKind": "K0smotronControlPlane", "worker count": 1}
2024-12-19T10:05:59Z    INFO    Starting workers        {"controller": "remotemachine", "controllerGroup": "infrastructure.cluster.x-k8s.io", "controllerKind": "RemoteMachine", "worker count": 1}
2024-12-19T10:05:59Z    INFO    Starting workers        {"controller": "k0sworkerconfig", "controllerGroup": "bootstrap.cluster.x-k8s.io", "controllerKind": "K0sWorkerConfig", "worker count": 1}
2024-12-19T10:05:59Z    INFO    Starting workers        {"controller": "cluster", "controllerGroup": "k0smotron.io", "controllerKind": "Cluster", "worker count": 1}
2024-12-19T10:05:59Z    INFO    Starting workers        {"controller": "remotecluster", "controllerGroup": "infrastructure.cluster.x-k8s.io", "controllerKind": "RemoteCluster", "worker count": 1}
2024-12-19T10:06:09Z    ERROR   controller-runtime.source.EventHandler  failed to get informer from cache       {"error": "failed to get API group resources: unable to retrieve the complete list of server APIs: cluster.x-k8s.io/v1beta1: the server could not find the requested resource"}
sigs.k8s.io/controller-runtime/pkg/internal/source.(*Kind).Start.func1.1
        /go/pkg/mod/sigs.k8s.io/controller-runtime@v0.16.5/pkg/internal/source/kind.go:68
k8s.io/apimachinery/pkg/util/wait.loopConditionUntilContext.func2
        /go/pkg/mod/k8s.io/apimachinery@v0.28.4/pkg/util/wait/loop.go:73
k8s.io/apimachinery/pkg/util/wait.loopConditionUntilContext
        /go/pkg/mod/k8s.io/apimachinery@v0.28.4/pkg/util/wait/loop.go:74
k8s.io/apimachinery/pkg/util/wait.PollUntilContextCancel
        /go/pkg/mod/k8s.io/apimachinery@v0.28.4/pkg/util/wait/poll.go:33
sigs.k8s.io/controller-runtime/pkg/internal/source.(*
sigs.k8s.io/controller-runtime/pkg/internal/source.(*Kind).Start.func1
        /go/pkg/mod/sigs.k8s.io/controller-runtime@v0.16.5/pkg/internal/source/kind.go:56
2024-12-19T10:06:29Z    ERROR   controller-runtime.source.EventHandler  failed to get informer from cache       {"error": "failed to get API group resources: unable to retrieve the complete list of server APIs: cluster.x-k8s.io/v1beta1: the server could not find the requested resource"}
sigs.k8s.io/controller-runtime/pkg/internal/source.(*Kind).Start.func1.1
        /go/pkg/mod/sigs.k8s.io/controller-runtime@v0.16.5/pkg/internal/source/kind.go:68
k8s.io/apimachinery/pkg/util/wait.loopConditionUntilContext.func2
        /go/pkg/mod/k8s.io/apimachinery@v0.28.4/pkg/util/wait/loop.go:73
k8s.io/apimachinery/pkg/util/wait.loopConditionUntilContext
        /go/pkg/mod/k8s.io/apimachinery@v0.28.4/pkg/util/wait/loop.go:74
k8s.io/apimachinery/pkg/util/wait.PollUntilContextCancel
        /go/pkg/mod/k8s.io/apimachinery@v0.28.4/pkg/util/wait/poll.go:33
sigs.k8s.io/controller-runtime/pkg/internal/source.(*Kind).Start.func1
        /go/pkg/mod/sigs.k8s.io/controller-runtime@v0.16.5/pkg/internal/source/kind.go:56
2024-12-19T10:06:39Z    ERROR   controller-runtime.source.EventHandler  failed to get informer from cache       {"error": "failed to get API group resources: unable to retrieve the complete list of server APIs: cluster.x-k8s.io/v1beta1: the server could not find the requested resource"}
sigs.k8s.io/controller-runtime/pkg/internal/source.(*Kind).Start.func1.1
        /go/pkg/mod/sigs.k8s.io/controller-runtime@v0.16.5/pkg/internal/source/kind.go:68
k8s.io/apimachinery/pkg/util/wait.loopConditionUntilContext.func2
        /go/pkg/mod/k8s.io/apimachinery@v0.28.4/pkg/util/wait/loop.go:73
k8s.io/apimachinery/pkg/util/wait.loopConditionUntilContext
        /go/pkg/mod/k8s.io/apimachinery@v0.28.4/pkg/util/wait/loop.go:74
k8s.io/apimachinery/pkg/util/wait.PollUntilContextCancel
        /go/pkg/mod/k8s.io/apimachinery@v0.28.4/pkg/util/wait/poll.go:33
sigs.k8s.io/controller-runtime/pkg/internal/source.(*Kind).Start.func1
        /go/pkg/mod/sigs.k8s.io/controller-runtime@v0.16.5/pkg/internal/source/kind.go:56
2024-12-19T10:06:49Z    ERROR   controller-runtime.source.EventHandler  failed to get informer from cache       {"error": "failed to get API group resources: unable to retrieve the complete list of server APIs: cluster.x-k8s.io/v1beta1: the server could not find the requested resource"}
sigs.k8s.io/controller-runtime/pkg/internal/source.(*Kind).Start.func1.1
        /go/pkg/mod/sigs.k8s.io/controller-runtime@v0.16.5/pkg/internal/source/kind.go:68
k8s.io/apimachinery/pkg/util/wait.loopConditionUntilContext.func2
        /go/pkg/mod/k8s.io/apimachinery@v0.28.4/pkg/util/wait/loop.go:73
k8s.io/apimachinery/pkg/util/wait.loopConditionUntilContext
        /go/pkg/mod/k8s.io/apimachinery@v0.28.4/pkg/util/wait/loop.go:74
k8s.io/apimachinery/pkg/util/wait.PollUntilContextCancel
        /go/pkg/mod/k8s.io/apimachinery@v0.28.4/pkg/util/wait/poll.go:33
sigs.k8s.io/controller-runtime/pkg/internal/source.(*Kind).Start.func1
        /go/pkg/mod/sigs.k8s.io/controller-runtime@v0.16.5/pkg/internal/source/kind.go:56
2024-12-19T10:06:59Z    ERROR   controller-runtime.source.EventHandler  failed to get
        /go/pkg/mod/sigs.k8s.io/controller-runtime@v0.16.5/pkg/internal/source/kind.go:56
2024-12-19T10:07:09Z    ERROR   controller-runtime.source.EventHandler  failed to get informer from cache       {"error": "failed to get API group resources: unable to retrieve the complete list of server APIs: cluster.x-k8s.io/v1beta1: the server could not find the requested resource"}
sigs.k8s.io/controller-runtime/pkg/internal/source.(*Kind).Start.func1.1
        /go/pkg/mod/sigs.k8s.io/controller-runtime@v0.16.5/pkg/internal/source/kind.go:68
k8s.io/apimachinery/pkg/util/wait.loopConditionUntilContext.func2
        /go/pkg/mod/k8s.io/apimachinery@v0.28.4/pkg/util/wait/loop.go:73
k8s.io/apimachinery/pkg/util/wait.loopConditionUntilContext
        /go/pkg/mod/k8s.io/apimachinery@v0.28.4/pkg/util/wait/loop.go:74
k8s.io/apimachinery/pkg/util/wait.PollUntilContextCancel
        /go/pkg/mod/k8s.io/apimachinery@v0.28.4/pkg/util/wait/poll.go:33
sigs.k8s.io/controller-runtime/pkg/internal/source.(*Kind).Start.func1
        /go/pkg/mod/sigs.k8s.io/controller-runtime@v0.16.5/pkg/internal/source/kind.go:56
2024-12-19T10:07:19Z    ERROR   controller-runtime.source.EventHandler  failed to get informer from cache       {"error": "failed to get API group resources: unable to retrieve the complete list of server APIs: cluster.x-k8s.io/v1beta1: the server could not find the requested resource"}
sigs.k8s.io/controller-runtime/pkg/internal/source.(*Kind).Start.func1.1
        /go/pkg/mod/sigs.k8s.io/controller-runtime@v0.16.5/pkg/internal/source/kind.go:68
k8s.io/apimachinery/pkg/util/wait.loopConditionUntilContext.func2
        /go/pkg/mod/k8s.io/apimachinery@v0.28.4/pkg/util/wait/loop.go:73
k8s.io/apimachinery/pkg/util/wait.loopConditionUntilContext
        /go/pkg/mod/k8s.io/apimachinery@v0.28.4/pkg/util/wait/loop.go:74
k8s.io/apimachinery/pkg/util/wait.PollUntilContextCancel
        /go/pkg/mod/k8s.io/apimachinery@v0.28.4/pkg/util/wait/poll.go:33
sigs.k8s.io/controller-runtime/pkg/internal/source.(*Kind).Start.func1
        /go/pkg/mod/sigs.k8s.io/controller-runtime@v0.16.5/pkg/internal/source/kind.go:56
2024-12-19T10:07:29Z    ERROR   controller-runtime.source.EventHandler  failed to get informer from cache       {"error": "failed to get API group resources: unable to retrieve the complete list of server APIs: cluster.x-k8s.io/v1beta1: the server could not find the requested resource"}
sigs.k8s.io/controller-runtime/pkg/internal/source.(*Kind).Start.func1.1
        /go/pkg/mod/sigs.k8s.io/controller-runtime@v0.16.5/pkg/internal/source/kind.go:68
k8s.io/apimachinery/pkg/util/wait.loopConditionUntilContext.func2
        /go/pkg/mod/k8s.io/apimachinery@v0.28.4/pkg/util/wait/loop.go:73
k8s.io/apimachinery/pkg/util/wait.loopConditionUntilContext
        /go/pkg/mod/k8s.io/apimachinery@v0.28.4/pkg/util/wait/loop.go:74
k8s.io/apimachinery/pkg/util/wait.PollUntilContextCancel
        /go/pkg/mod/k8s.io/apimachinery@v0.28.4/pkg/util/wait/poll.go:33
sigs.k8s.io/controller-runtime/pkg/internal/source.(*Kind).Start.func1
        /go/pkg/mod/sigs.k8s.io/controller-runtime@v0.16.5/pkg/internal/source/kind.go:56
2024-12-19T10:07:39Z    ERROR   controller-runtime.source.EventHandler  failed to get informer from cache       {"error": "failed to get API group resources: unable to retrieve the complete list of server APIs: cluster.x-k8s.io/v1beta1: the server could not find the requeste
ources: unable to retrieve the complete list of server APIs: cluster.x-k8s.io/v1beta1: the server could not find the requested resource"}
sigs.k8s.io/controller-runtime/pkg/internal/source.(*Kind).Start.func1.1
        /go/pkg/mod/sigs.k8s.io/controller-runtime@v0.16.5/pkg/internal/source/kind.go:68
k8s.io/apimachinery/pkg/util/wait.loopConditionUntilContext.func2
        /go/pkg/mod/k8s.io/apimachinery@v0.28.4/pkg/util/wait/loop.go:73
k8s.io/apimachinery/pkg/util/wait.loopConditionUntilContext
        /go/pkg/mod/k8s.io/apimachinery@v0.28.4/pkg/util/wait/loop.go:74
k8s.io/apimachinery/pkg/util/wait.PollUntilContextCancel
        /go/pkg/mod/k8s.io/apimachinery@v0.28.4/pkg/util/wait/poll.go:33
sigs.k8s.io/controller-runtime/pkg/internal/source.(*Kind).Start.func1
        /go/pkg/mod/sigs.k8s.io/controller-runtime@v0.16.5/pkg/internal/source/kind.go:56
2024-12-19T10:07:59Z    ERROR   controller-runtime.source.EventHandler  failed to get informer from cache       {"error": "failed to get API group resources: unable to retrieve the complete list of server APIs: cluster.x-k8s.io/v1beta1: the server could not find the requested resource"}
sigs.k8s.io/controller-runtime/pkg/internal/source.(*Kind).Start.func1.1
        /go/pkg/mod/sigs.k8s.io/controller-runtime@v0.16.5/pkg/internal/source/kind.go:68
k8s.io/apimachinery/pkg/util/wait.loopConditionUntilContext.func2
        /go/pkg/mod/k8s.io/apimachinery@v0.28.4/pkg/util/wait/loop.go:73
k8s.io/apimachinery/pkg/util/wait.loopConditionUntilContext
        /go/pkg/mod/k8s.io/apimachinery@v0.28.4/pkg/util/wait/loop.go:74
k8s.io/apimachinery/pkg/util/wait.PollUntilContextCancel
        /go/pkg/mod/k8s.io/apimachinery@v0.28.4/pkg/util/wait/poll.go:33
sigs.k8s.io/controller-runtime/pkg/internal/source.(*Kind).Start.func1
        /go/pkg/mod/sigs.k8s.io/controller-runtime@v0.16.5/pkg/internal/source/kind.go:56
2024-12-19T10:07:59Z    ERROR   Could not wait for Cache to sync        {"controller": "k0scontrolplane", "controllerGroup": "controlplane.cluster.x-k8s.io", "controllerKind": "K0sControlPlane", "error": "failed to wait for k0scontrolplane caches to sync: timed out waiting for cache to be synced for Kind *v1beta1.Machine"}
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.1
        /go/pkg/mod/sigs.k8s.io/controller-runtime@v0.16.5/pkg/internal/controller/controller.go:203
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2
        /go/pkg/mod/sigs.k8s.io/controller-runtime@v0.16.5/pkg/internal/controller/controller.go:208
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start
        /go/pkg/mod/sigs.k8s.io/controller-runtime@v0.16.5/pkg/internal/controller/controller.go:234
sigs.k8s.io/controller-runtime/pkg/manager.(*runnableGroup).reconcile.func1
        /go/pkg/mod/sigs.k8s.io/controller-runtime@v0.16.5/pkg/manager/runnable_group.go:223
2024-12-19T10:07:59Z    INFO    Stopping and waiting for non leader election runnables
2024-12-19T10:07:59Z    INFO    Stopping and waiting for leader election runnables
2024-12-19T10:07:59Z    INFO    Shutdown signal received, waiting for all workers to finish     {"controller": "remotecluster", "controllerGroup": "infrastructure.cluster.x-k8s.io", "controllerKind": "RemoteCluster"}
2024-12-19T10:07:59Z    INFO    Shutdown signal received, waiting for all workers to finish     {"controller": "cluster", "controllerGroup": "k0smotron.io", "controllerKind": "Cluster"}
2024-12-19T10:07:59Z    INFO    Shutdown signal received, waiting for all workers to finish     {"controller": "k0sworkerconfig", "controllerGroup": "bootstrap.cluster.x-k8s.io", "controllerKind": "K0sWorkerConfig"}
2024-12-19T10:07:59Z    INFO    Shutdown signal received, waiting for all workers to finish     {"controller": "remotemachine", "controllerGroup": "infrastructure.cluster.x-k8s.io", "controllerKind": "RemoteMachine"}
2024-12-19T10:07:59Z    INFO    Shutdown signal received, waiting for all workers to finish     {"controller": "k0smotroncontrolplane", "controllerGroup": "controlplane.cluster.x-k8s.io", "controllerKind": "K0smotronControlPlane"}
2024-12-19T10:07:59Z    INFO    Shutdown signal received, waiting for all workers to finish     {"controller": "k0scontrollerconfig", "controllerGroup": "bootstrap.cluster.x-k8s.io", "controllerKind": "K0sControllerConfig"}
2024-12-19T10:07:59Z    INFO    Shutdown signal received, waiting for all workers to finish     {"controller": "jointokenrequest", "controllerGroup": "k0smotron.io", "controllerKind": "JoinTokenRequest"}
2024-12-19T10:07:59Z    INFO    All workers finished    {"controller": "remotecluster", "controllerGroup": "infrastructure.cluster.x-k8s.io", "controllerKind": "RemoteCluster"}
2024-12-19T10:07:59Z    INFO    All workers finished    {"controller": "jointokenrequest", "controllerGroup": "k0smotron.io", "controllerKind": "JoinTokenRequest"}
2024-12-19T10:07:59Z    INFO    All workers finished    {"controller": "k0scontrollerconfig", "controllerGroup": "bootstrap.cluster.x-k8s.io", "controllerKind": "K0sControllerConfig"}
2024-12-19T10:07:59Z    INFO    All workers finished    {"controller": "k0smotroncontrolplane", "controllerGroup": "controlplane.cluster.x-k8s.io", "controllerKind": "K0smotronControlPlane"}
2024-12-19T10:07:59Z    INFO    All workers finished    {"controller": "k0sworkerconfig", "controllerGroup": "bootstrap.cluster.x-k8s.io", "controllerKind": "K0sWorkerConfig"}
2024-12-19T10:07:59Z    INFO    All workers finished    {"controller": "remotemachine", "controllerGroup": "infrastructure.cluster.x-k8s.io", "controllerKind": "RemoteMachine"}
2024-12-19T10:07:59Z    INFO    All workers finished    {"controller": "cluster", "controllerGroup": "k0smotron.io", "controllerKind": "Cluster"}
2024-12-19T10:07:59Z    INFO    Stopping and waiting for caches
2024-12-19T10:07:59Z    INFO    Stopping and waiting for webhooks
2024-12-19T10:07:59Z    INFO    Stopping and waiting for HTTP servers
2024-12-19T10:07:59Z    INFO    shutting down server    {"kind": "health probe", "addr": "[::]:8081"}
2024-12-19T10:07:59Z    INFO    controller-runtime.metrics      Shutting down metrics server with timeout of 1 minute
2024-12-19T10:07:59Z    INFO    Wait completed, proceeding to shutdown the manager
2024-12-19T10:07:59Z    ERROR   setup   problem running manager {"error": "failed to wait for k0scontrolplane caches to sync: timed out waiting for cache to be synced for Kind *v1beta1.Machine"}
main.main
        /workspace/cmd/main.go:246
runtime.main
        /usr/local/go/src/runtime/proc.go:271
```

**Comment by makhov** [comment]
@EKami yeah, sorry, the same happened in v1.2.0. We've just published v1.3.0 with the fix

**Comment by EKami** [comment]
> [@EKami](https://github.com/EKami) yeah, sorry, the same happened in v1.2.0. We've just published v1.3.0 with the fix

No worries, thanks for the reply, I'll stick to 1.1.2 in that case until 1.3.0 releases.
